### PR TITLE
204 every new account gets a randomized username

### LINKED ISSUE #204: Every new account gets a randomized username
# Notion
https://www.notion.so/citizenwallet/Every-new-account-gets-a-randomized-username-7f4d622004914326939a8d622157c7fc?pvs=4

### PR DESCRIPTION
**Pull request by SajeeTechi**
Closes #204

### CODE REVIEW

**Review by kevtechi — CHANGES_REQUESTED** [review]
Feedback given during a call.

For migration of contacts:
1. there should be a json profile in the table
2. take data from there or don't migrate the row

For fetching a profile:
1. get from db
3. if not available locally >> await fetch from profile contract (add to db)
4. if available locally >> return immediately (async update from profile contract)

File: lib/services/db/backup/accounts.dart [hunk 1]
```diff
@@ -92,12 +125,48 @@ class AccountsTable extends DBTable {
...
+          } catch (e, s) {
+            debugPrint('$name migration error, index $i: $e');
+            debugPrintStack(stackTrace: s);
+          }
         }
       }
     }
   }
 
+    Future<void> _migrateProfiles(Database db) async {
```
Comment (kevtechi) [inline lib/services/db/backup/accounts.dart:143]: this migration is not needed. Every time an account is opened, we go to the wallet screen and go through the profile async loading flow which will load or create one

File: lib/services/db/backup/accounts.dart [hunk 2]
```diff
@@ -79,10 +106,16 @@ class AccountsTable extends DBTable {
 
   @override
   Future<void> migrate(Database db, int oldVersion, int newVersion) async {
-    final migrations = {     
+    final migrations = {
       2: [
         'UPDATE $name SET privateKey = NULL',
       ],
+      3: [
+        'ALTER TABLE $name ADD COLUMN username TEXT DEFAULT NULL',
```
Comment (kevtechi) [inline lib/services/db/backup/accounts.dart:114]: you only need the table column migration

File: lib/state/profile/logic.dart [hunk 3]
```diff
@@ -187,6 +194,15 @@ class ProfileLogic {
         profile,
```
Comment (kevtechi) [inline lib/state/profile/logic.dart:194]: upsert as well from the db fetch

**Review by kevtechi — CHANGES_REQUESTED** [review]

File: lib/state/profile/logic.dart [hunk 4]
```diff
@@ -382,4 +420,114 @@ class ProfileLogic {
...
+          name: newProfile.name,
+          description: newProfile.description,
+          image: newProfile.image,
+          imageMedium: newProfile.imageMedium,
+          imageSmall: newProfile.imageSmall,
+        ),
+      );
+
+      _accountBackupDBService.accounts.update(
+        DBAccount(
```
[the remark this answers is not in the document]
Comment (SajeeTechi) [reply lib/state/profile/logic.dart:520]: i am updating the values `name`, `username` and `profile`

`alias` and `address` are required to create the unique identifier for the update query

**Review by kevtechi — APPROVED** [review]

**Pull request merged**